### PR TITLE
WRN-1380: Migrate Dropdown unit tests to Testing Library

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Migrated unit tests from Enzyme to Testing Library for Dropdown component.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-1380

### CODE REVIEW

File: Dropdown/tests/Dropdown-specs.js [hunk 1]
```diff
@@ -1,261 +1,271 @@
...
+			<Dropdown title={title}>
+				{updatedChildren}
+			</Dropdown>
+		);
 
 		const expected = 5;
-		const actual = dropDown.children().length;
+		const actual = updatedChildren.length;
 
 		expect(actual).toBe(expected);
```
Comment (alexandrumorariu) [inline Dropdown/tests/Dropdown-specs.js:185]: Here you can use toHaveLength() instead, but it is up to you.

File: Dropdown/tests/Dropdown-specs.js [hunk 2]
```diff
@@ -1,261 +1,271 @@
...
+
+		rerender(
+			<Dropdown title={title}>
+				{updatedChildren}
+			</Dropdown>
+		);
 
 		const expected = 5;
-		const actual = dropDown.children().length;
+		const actual = updatedChildren.length;
```
Comment (daniel-stoian-lgp) [inline Dropdown/tests/Dropdown-specs.js:183]: Unfortunately, this does not verify that the actual dropdown has 5 options, but just that an array is updated. You will proably need to change the test to render an opened dropdown and check the number of children of a node(something like screen.getByRole("list").children.length

Comment (stanca-pop-lgp) [reply Dropdown/tests/Dropdown-specs.js:183]: You are right. Thank you for the suggestion!

**Review by alexandrumorariu — APPROVED** [review]
LGTM

**Review by daniel-stoian-lgp — APPROVED** [review]

**Pull request merged**